code/modules/transport/pods/MainWeapon.dm

```diff
@@ -380,6 +380,563 @@
 		opencomputer(usr)
 		return
 
+// Extra-Facility Integrity Fabricator: the big boyo pod construction tool
+
+//mode defines for tool
+#define EFIF_MODE_DRILL 0
+#define EFIF_MODE_FLOORS 1
+#define EFIF_MODE_R_FLOORS 2
+#define EFIF_MODE_WALLS 3
+#define EFIF_MODE_REPAIR 4
+
+//construction time modifier per tile
+#define EFIF_FLOOR_BUILD_TIME 2
+#define EFIF_R_FLOOR_BUILD_TIME 3
+#define EFIF_WALL_BUILD_TIME 6
+//walls will also require the floor build time if built directly on space
+
+//construction cost per tile
+#define EFIF_FLOOR_COST 1
+#define EFIF_R_FLOOR_COST 2
+#define EFIF_WALL_COST 4
+
+TYPEINFO(/obj/item/shipcomponent/mainweapon/constructor)
+	mats = list("metal_superdense" = 50, "claretine" = 20, "electrum" = 10)
+
+/obj/item/shipcomponent/mainweapon/constructor
+	name = "EFIF-1 Construction System"
+	desc = "A pair of elaborate robotic arms equipped for large-scale construction and asteroid demolition."
+	current_projectile = new/datum/projectile/laser/drill/cutter
+	appearanceString = "pod_weapon_efif"
+	icon_state = "constructor"
+	firerate = 12
+	var/mode = EFIF_MODE_DRILL
+	///Current count of loaded steel sheets (only accepts steel, as the system's metalforming is designed for it)
+	var/steel_sheets = 0
```

Comment:
start with 20 sheets?

Reply:
Since it can be manufactured, I'd lean towards no for the baseline version. I could make a "loaded" subpath for the cargo purchase that starts with 100 or 200, though.

Reply:
could roll the steel cost into the fabrication cost, since you will (almost always) want sheets in this?

Reply:
Zero initial sheets means 4 regular-sized stacks or 1 QM-ordered megastack will cleanly fit into the device, and also, requiring 5+ steel bars would usually just mean running sheets through the reclaimer to then... turn them back into sheets again.

Reply:
QM version is now prestocked with 200 sheets.